### PR TITLE
Publishing vault status on ZFS platforms

### PR DESCRIPTION
**Pull request by cshari-zededa**
Signed-off-by: Hariharasubramanian C S <cshari@zededa.com>

### CODE REVIEW

**Comment by rvs** [comment]
Hey @eriknordmark -- since 5.9.0 is tagged -- please take a look and pull if it looks OK to you.

File: pkg/pillar/cmd/vaultmgr/vaultmgr.go [hunk 1]
```diff
@@ -479,33 +479,61 @@ func initializeSelfPublishHandles(ps *pubsub.PubSub, ctx *vaultMgrContext) {
 	ctx.pubVaultStatus = pubVaultStatus
 }
 
-//GetOperInfo gets the current operational state of fscrypt. (Deprecated)
-func GetOperInfo() (info.DataSecAtRestStatus, string) {
+//getFscryptOperInfo does some basic fscrypt checks
```
Comment (eriknordmark) [inline pkg/pillar/cmd/vaultmgr/vaultmgr.go:482]: I still just returns the info, hence the "checks" part of the comment seems misleading.

File: pkg/pillar/cmd/vaultmgr/vaultmgr.go [hunk 2]
```diff
@@ -479,33 +479,61 @@ func initializeSelfPublishHandles(ps *pubsub.PubSub, ctx *vaultMgrContext) {
...
-			return info.DataSecAtRestStatus_DATASEC_AT_REST_DISABLED,
-				"File system is incompatible, needs a disruptive upgrade"
-		}
+		return info.DataSecAtRestStatus_DATASEC_AT_REST_ERROR,
+			"Fscrypt is not enabled"
+
+	}
+}
+
+//getZfsOperInfo does some basic zfs checks
```
Comment (eriknordmark) [inline pkg/pillar/cmd/vaultmgr/vaultmgr.go:504]: Ditto

File: pkg/pillar/cmd/vaultmgr/vaultmgr.go (hunk 1 above)
Comment (cshari-zededa) [reply pkg/pillar/cmd/vaultmgr/vaultmgr.go]: changed it

File: pkg/pillar/cmd/vaultmgr/vaultmgr.go (hunk 2 above)
Comment (cshari-zededa) [reply pkg/pillar/cmd/vaultmgr/vaultmgr.go]: changed it

**Pull request merged**